src/components/StorageOverview/TopConsumers/TopConsumers.js

```diff
@@ -92,37 +98,160 @@ const TopConsumersBody = ({ topConsumers }) => {
   return <div>{results}</div>;
 };
 
-export const TopConsumers = ({ topConsumers, LoadingComponent }) => (
-  <DashboardCard>
-    <DashboardCardHeader>
-      <DashboardCardTitle>Top Projects by Requested Capacity</DashboardCardTitle>
-    </DashboardCardHeader>
-    <DashboardCardBody isLoading={!topConsumers} LoadingComponent={LoadingComponent}>
-      <TopConsumersBody topConsumers={topConsumers} />
-    </DashboardCardBody>
-  </DashboardCard>
-);
+export class TopConsumers extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: metricTypes[0],
+      sortBy: sortBy[0],
+    };
+  }
 
-TopConsumersBody.defaultProps = {
-  topConsumers: null,
-};
+  changeState = (key, newVal) => {
+    this.setState({ [key]: newVal }, this.getCurrentMetric);
+  };
+
+  getCapacity = data => get(data, 'data.result', []);
+
+  getCurrentMetric = () => {
+    switch (this.state.type) {
+      case PROJECTS:
+        switch (this.state.sortBy) {
+          case BY_USED_CAPACITY:
+            return this.getCapacity(this.props.projectsUsedCapacity);
```

Comment:
maybe object mapping (of sortBy to project prop name, etc.) would be easier to read than these many switches?

Reply:
@suomiy will also take this up in the follow up PR.